nan.h

```diff
@@ -605,31 +605,45 @@ class NanEscapableScope {
     v8::Isolate::GetCurrent()->ThrowException(error);
   }
 
-  NAN_INLINE v8::Local<v8::Object> NanNewBufferHandle (
+  NAN_INLINE NanMaybeLocal<v8::Object> NanNewBuffer(
       char *data
     , size_t length
     , node::smalloc::FreeCallback callback
     , void *hint
   ) {
-    return node::Buffer::New(
-        v8::Isolate::GetCurrent(), data, length, callback, hint);
+    // arbitrary buffer lengths requires
+    // NODE_MODULE_VERSION >= IOJS_3_0_MODULE_VERSION
+    assert(length <= 0x3fffffff && "too large buffer");
+    return NanMaybeLocal<v8::Object>(node::Buffer::New(
+        v8::Isolate::GetCurrent(), data, length, callback, hint));
   }
 
-  NAN_INLINE v8::Local<v8::Object> NanNewBufferHandle (
+  NAN_INLINE NanMaybeLocal<v8::Object> NanCopyBuffer(
       const char *data
     , uint32_t size
   ) {
-    return node::Buffer::New(v8::Isolate::GetCurrent(), data, size);
+    // arbitrary buffer lengths requires
+    // NODE_MODULE_VERSION >= IOJS_3_0_MODULE_VERSION
+    assert(size <= 0x3fffffff && "too large buffer");
+    return NanMaybeLocal<v8::Object>(node::Buffer::New(
+        v8::Isolate::GetCurrent(), data, size));
   }
 
-  NAN_INLINE v8::Local<v8::Object> NanNewBufferHandle (uint32_t size) {
-    return node::Buffer::New(v8::Isolate::GetCurrent(), size);
+  NAN_INLINE NanMaybeLocal<v8::Object> NanNewBuffer(uint32_t size) {
+    // arbitrary buffer lengths requires
+    // NODE_MODULE_VERSION >= IOJS_3_0_MODULE_VERSION
+    assert(size <= 0x3fffffff && "too large buffer");
+    return NanMaybeLocal<v8::Object>(node::Buffer::New(
+        v8::Isolate::GetCurrent(), size));
   }
 
-  NAN_INLINE v8::Local<v8::Object> NanBufferUse(
+  NAN_INLINE v8::Local<v8::Object> NanNewBuffer(
       char* data
     , uint32_t size
   ) {
+    // arbitrary buffer lengths requires
+    // NODE_MODULE_VERSION >= IOJS_3_0_MODULE_VERSION
+    assert(size <= 0x3fffffff && "too large buffer");
     return node::Buffer::Use(v8::Isolate::GetCurrent(), data, size);
```

Comment:
I'm assuming this one cannot return an empty handle, as it should not be calling malloc.